tests/test_blueprint.py

```diff
@@ -95,13 +96,36 @@ def test_messages_channel(bp, app, mockredis):
             "data": '{"data": "whee", "id": "abc"}',
         }
     ]
+    pubsub.unsubscribe.side_effect = redis.exceptions.ConnectionError()
 
     gen = bp.messages('whee')
 
     assert isinstance(gen, types.GeneratorType)
     output = list(gen)
     assert output == [flask_sse.Message("whee", id="abc")]
     pubsub.subscribe.assert_called_with('whee')
+    pubsub.unsubscribe.assert_called_with('whee')
+
+
+def test_messages_close(bp, app, mockredis):
```

Comment:
Thanks for updating the existing tests, and adding this one!

Reply:
You're welcome! Was easy to do.